k2/csrc/array_ops_test.cu

```diff
@@ -1069,4 +1069,252 @@ TEST(OpsTest, SpliceRowSplitsTest) {
+                                                 12, 13, 15, 15, 16};
+    const std::vector<int32_t> row_ids_vec = {0, 0, 1, 2, 2, 3, 3, 3,
+                                              4, 5, 5, 5, 6, 7, 7, 9};
+    Array1<int32_t> row_ids(context, row_ids_vec);
+    Array1<int32_t> row_splits(context, row_splits_vec);
+    EXPECT_FALSE(ValidateRowSplits(row_splits));
+    EXPECT_TRUE(ValidateRowIds(row_ids));
+    EXPECT_FALSE(ValidateRowSplitsAndIds(row_splits, row_ids));
+  }
+
+  {
+    // bad case row ids, contains negative value
+    const std::vector<int32_t> row_splits_vec = {0,  2,  3,  5,  8, 9,
+                                                 12, 13, 15, 15, 16};
+    const std::vector<int32_t> row_ids_vec = {0, 0, 1, 2, -2, 3, 3, 3,
+                                              4, 5, 5, 5, 6,  7, 7, 9};
+    Array1<int32_t> row_ids(context, row_ids_vec);
+    Array1<int32_t> row_splits(context, row_splits_vec);
+    EXPECT_TRUE(ValidateRowSplits(row_splits));
+    EXPECT_FALSE(ValidateRowIds(row_ids));
+    EXPECT_FALSE(ValidateRowSplitsAndIds(row_splits, row_ids));
+  }
+
+  {
+    // bad case row ids, not non-decreasing
+    const std::vector<int32_t> row_splits_vec = {0,  2,  3,  5,  8, 9,
+                                                 12, 13, 15, 15, 16};
+    const std::vector<int32_t> row_ids_vec = {0, 0, 1, 2, 2, 3, 3, 3,
+                                              4, 5, 5, 5, 6, 7, 6, 9};
+    Array1<int32_t> row_ids(context, row_ids_vec);
+    Array1<int32_t> row_splits(context, row_splits_vec);
+    EXPECT_TRUE(ValidateRowSplits(row_splits));
+    EXPECT_FALSE(ValidateRowIds(row_ids));
+    EXPECT_FALSE(ValidateRowSplitsAndIds(row_splits, row_ids));
+  }
+
+  {
+    // bad case row ids and row splits don't agree with each other
+    // i < row_splits[row_ids[i]]
+    const std::vector<int32_t> row_splits_vec = {0,  2,  3,  5,  8, 9,
+                                                 12, 13, 15, 15, 16};
+    const std::vector<int32_t> row_ids_vec = {0, 0, 1, 2, 2, 3, 3, 3,
+                                              4, 5, 5, 5, 6, 7, 8, 9};
+    Array1<int32_t> row_ids(context, row_ids_vec);
+    Array1<int32_t> row_splits(context, row_splits_vec);
+    EXPECT_TRUE(ValidateRowSplits(row_splits));
+    EXPECT_TRUE(ValidateRowIds(row_ids));
+    EXPECT_FALSE(ValidateRowSplitsAndIds(row_splits, row_ids));
+  }
+
+  {
+    // another bad case that row ids and row splits don't agree with each other
+    // i > = row_splits[row_ids[i]]
+    const std::vector<int32_t> row_splits_vec = {0,  2,  3,  5,  8, 9,
+                                                 12, 13, 15, 15, 16};
+    const std::vector<int32_t> row_ids_vec = {0, 0, 1, 2, 2, 3, 3, 3,
+                                              4, 5, 5, 5, 5, 7, 7, 9};
+    Array1<int32_t> row_ids(context, row_ids_vec);
+    Array1<int32_t> row_splits(context, row_splits_vec);
+    EXPECT_TRUE(ValidateRowSplits(row_splits));
```

Comment:
As `temp` is not used here, should we define it inside the function instead of passing
it as an argument?

Reply:
Not fix for now, see my explanation above